DatabaseManager.py

```diff
@@ -0,0 +1,85 @@
+import sqlite3
+import logging
+import pandas as pd
+from typing import Dict
+from pathlib import Path
+import os
+
+class DatabaseManager:
+    def __init__(self, db_path: str = "spotify_db/spotify.db", schema_dir: str = "spotify_db/schema"):
+        self.db_path = db_path
+        self.schema_dir = schema_dir
+        self.connection = None
+        self._initialize_database()
+
+    def _load_schema_file(self, filename: str) -> str:
+        schema_path = Path(self.schema_dir) / filename
+        with open(schema_path, 'r', encoding='utf-8') as f:
+            return f.read()
+
+    def _initialize_database(self):
+        if Path(self.db_path).exists():
+            os.remove(self.db_path)
+
+        self.connection = sqlite3.connect(self.db_path)
+        cursor = self.connection.cursor()
+        cursor.execute("PRAGMA foreign_keys = OFF")
+
+        schema_dir_path = Path(self.schema_dir)
+        sql_files = sorted(schema_dir_path.glob("*.sql"))
+
+        for schema_file in sql_files:
+            sql = self._load_schema_file(schema_file)
+            cursor.executescript(sql)
+
+        self.connection.commit()
+
+    def save_data(self, dataframes: Dict[str, pd.DataFrame]) -> bool:
+        if not self.connection:
+            logging.error("Database connection not established")
+            return False
+
+        try:
+            cursor = self.connection.cursor()
+            cursor.execute("PRAGMA foreign_keys = OFF")
```

Comment:
Unless you’re seeing foreign-key errors during your normal load, remove the `PRAGMA foreign_keys = OFF` line. Don't need to overcomplicate things